scripts/mergeDBSamples.py

```diff
@@ -0,0 +1,259 @@
+    return gitHash, repo, url
+
+def add_merged_sample(NAME, type, AnaUrl, FWUrl, samples):
+    # samples is a simple dict containing three keys: 'process', 'dataset_id', 'sample_id'
+    dbstore = DbStore()
+    sample = None
+
+    # check that source dataset exist
+    # Skip: should exist, the check has been done before calling this function
+
+    # check that there is no existing entry
+    update = False
+    localpath = ''
+    nevents = 0
+    checkExisting = dbstore.find(Sample, Sample.name == unicode(NAME))
+    if checkExisting.is_empty():
+        sample = Sample(unicode(NAME), unicode(localpath), unicode(type), nevents)
+    else:
+        update = True
+        sample = checkExisting.one()
+        sample.removeFiles(dbstore)
+
+    # collecting contents
+    sample.nevents_processed = 0
+    sample.nevents = 0
+    sample.normalization = 1
+    sample.event_weight_sum = 0
+    extras_event_weight_sum = {}
+    dataset_nevents = 0
+    processed_lumi = LumiList()
+    for i, s in enumerate(samples):
+        if i == 0:
+            sample.source_dataset_id = s['dataset_id']
+            sample.source_sample_id = s['sample_id']
+        results = dbstore.find(Sample, Sample.sample_id == s['sample_id'])
+        # Should exist, the check has been done before calling this function
+        sample.nevents_processed += results[0].nevents_processed
+        sample.nevents += results[0].nevents
+        sample.event_weight_sum += results[0].event_weight_sum
+        extra_sumw = results[0].extras_event_weight_sum
+        if extra_sumw is not None:
+            extra_sumw = json.loads(extra_sumw)
+            for key in extra_sumw:
+                try:
+                    extras_event_weight_sum[key] += extra_sumw[key]
+                except KeyError:
+                    extras_event_weight_sum[key] = extra_sumw[key]
+        tmp_processed_lumi = results[0].processed_lumi
+        if tmp_processed_lumi is not None:
+            tmp_processed_lumi = json.loads( tmp_processed_lumi )
+            processed_lumi = processed_lumi | LumiList(compactList = tmp_processed_lumi)
+        # Get info from file table
+        results = dbstore.find(File, File.sample_id == s['sample_id'])
+        for lfn, pfn, event_weight_sum, file_extras_event_weight_sum, nevents in list(results.values(File.lfn, File.pfn, File.event_weight_sum, File.extras_event_weight_sum, File.nevents)):
+            f = File(lfn, pfn, event_weight_sum, file_extras_event_weight_sum, nevents)
+            sample.files.add(f)
+        # Get info from parent datasets
+        results = dbstore.find(Dataset, Dataset.dataset_id == s['dataset_id'])
+        dataset_nevents +=  results[0].nevents
+    if len(extras_event_weight_sum) > 0:
+        sample.extras_event_weight_sum = unicode(extras_event_weight_sum)
+    if len(processed_lumi.getCompactList()) > 0:
+        sample.processed_lumi = unicode(processed_lumi.getCompactList())
```

Comment:
This must be encoded into JSON, a python representation is not valid. It should be:

```python
unicode(json.dumps(processed_lumi.getCompactList()))
```